src/sage/combinat/specht_module.py

```diff
@@ -18,15 +25,227 @@
 from sage.rings.rational_field import QQ
-from sage.modules.with_basis.subquotient import SubmoduleWithBasis
+from sage.modules.with_basis.subquotient import SubmoduleWithBasis, QuotientModuleWithBasis
+from sage.modules.free_module_element import vector
 from sage.categories.modules_with_basis import ModulesWithBasis
 
+class SymmetricGroupRepresentation:
+    """
+    Mixin class for symmetric group (algebra) representations.
+    """
+    def __init__(self, SGA):
+        """
+        Initialize ``self``.
+
+        EXAMPLES::
+
+            sage: SM = Partition([3,1,1]).specht_module(GF(3))
+            sage: TestSuite(SM).run()
+        """
+        self._semigroup = SGA.group()
+        self._SGA = SGA
+
+    @cached_method
+    def frobenius_image(self):
+        r"""
+        Return the Frobenius image of ``self``.
+
+        The Frobenius map is defined as the map to symmetric functions
+
+        .. MATH::
+
+            F(\chi) = \frac{1}{n!} \sum_{w \in S_n} \chi(w) p_{\rho(w)},
+
+        where `\chi` is the character of the `S_n`-module ``self``,
+        `p_{\lambda}` is the powersum symmetric function basis element
+        indexed by `\lambda`, and `\rho(w)` is partition of the cycle type
+        of `w`. Specifically, this map takes irreducible representations
+        indexed by `\lambda` to the Schur function `s_{\lambda}`.
+
+        EXAMPLES::
+
+            sage: s = SymmetricFunctions(QQ).s()
+            sage: SM = Partition([2,2,1]).specht_module(QQ)
+            sage: s(SM.frobenius_image())
+            s[2, 2, 1]
+            sage: SM = Partition([4,1]).specht_module(CyclotomicField(5))
+            sage: s(SM.frobenius_image())
+            s[4, 1]
+
+        We verify the regular representation::
+
+            sage: from sage.combinat.diagram import Diagram
+            sage: D = Diagram([(0,0), (1,1), (2,2), (3,3), (4,4)])
+            sage: F = s(D.specht_module(QQ).frobenius_image()); F
+            s[1, 1, 1, 1, 1] + 4*s[2, 1, 1, 1] + 5*s[2, 2, 1]
+             + 6*s[3, 1, 1] + 5*s[3, 2] + 4*s[4, 1] + s[5]
+            sage: F == sum(StandardTableaux(la).cardinality() * s[la]
+            ....:          for la in Partitions(5))
+            True
+            sage: all(s[la] == s(la.specht_module(QQ).frobenius_image())
+            ....:     for n in range(1, 5) for la in Partitions(n))
+            True
+
+            sage: D = Diagram([(0,0), (1,1), (1,2), (2,3), (2,4)])
+            sage: SM = D.specht_module(QQ)
+            sage: s(SM.frobenius_image())
+            s[2, 2, 1] + s[3, 1, 1] + 2*s[3, 2] + 2*s[4, 1] + s[5]
+
+        An example using the tabloid module::
+
+            sage: SGA = SymmetricGroupAlgebra(QQ, 5)
+            sage: TM = SGA.tabloid_module([2, 2, 1])
+            sage: s(TM.frobenius_image())
+            s[2, 2, 1] + s[3, 1, 1] + 2*s[3, 2] + 2*s[4, 1] + s[5]
+        """
+        from sage.combinat.sf.sf import SymmetricFunctions
+        BR = self.base_ring()
```

Comment:
Are you sure you want ``self.base_ring()`` instead of ``QQ``? I can easily find myself applying this to Spechts over `\ZZ`, and the way you compute this using the p-basis will probably not work there.

Reply:
I guess this is a point I am slightly confused about with the Frobenius map. What is the image supposed to be? I am pretty sure you are correct with having this always be `QQ`, but is that what should happen over $F_p$ (assuming we just map irreducibles)? I don't know the precise details enough to answer this...

Reply:
I would be quite surprised if there was a Frobenius characteristic map over `GF(2)`. After all, there is a Specht module over `GF(2)` that decomposes as a direct sum of two Specht modules (see Exercise 1 (d) in https://www.ma.rhul.ac.uk/~uvah099/Maths/Sym/SymProblems2.pdf ), but no such identity holds for Schur functions even over `GF(2)`. Maybe there are some targets better than `QQ`, but `QQ` is at the very least a good start.

Reply:
Indeed, the modular branching rule is complicated (and one of the reasons for implementing this). Perhaps this should simply be disabled unless in char 0. There are definitely reasons for allowing more generic rings (mainly polynomial rings to compute the graded multiplicities). However, for the current implementations, it is either irreducible (hence, concentrated in a single degree) nor do I know of a generic grading for the general diagrams. So everything would be ungraded. I am now convinced to do this only over `QQ`.

Something else I have been pondering is to just return as Schur functions (of course, going through the powersums for the computation). This is the most useful output, but it is not the "natural" one from the computation. What do you think?

Reply:
Changed to `QQ` with also converting the coefficients for the powersums.